trivy-image-scan/action.yml

```diff
@@ -0,0 +1,76 @@
+name: Trivy Image Scan
+description: 'Trivy Image Scan'
+inputs:
+  image:
+    description: 'image name'
+    required: true
+  tag:
+    description: 'image tag'
+    required: false
+  ignore-unfixed:
+    description: 'ignore unfixed vulnerabilities'
+    required: true
+    default: 'false'
+  vuln-type:
+    description: 'comma-separated list of vulnerability types (os,library)'
+    required: true
+    default: 'os,library'
+  severity:
+    description: 'severities of vulnerabilities to be displayed'
+    required: true
+    default: 'HIGH,CRITICAL'
+  output:
+    description: 'writes results to a file with the specified file name'
+    required: true
+    default: 'trivy-results.sarif'
+  timeout:
+    description: 'timeout (default 5m0s)'
+    required: true
+    default: '5m0s'
+  trivyignores:
+    description: 'comma-separated list of relative paths in repository to one or more .trivyignore files'
+    required: true
+    default: '.trivyignore'
+  format:
+    description: 'output format (table, json, template)'
+    required: true
+    default: 'sarif'
+runs:
+  using: "composite"
+  steps:
+    - name: Determine container tag
+      if: "${{ github.event.inputs.tag == '' }}"
+      id: tag
+      run: echo tag=$(./gradlew -q printDockerImageDefaultTag | head -1) >> $GITHUB_OUTPUT
+
+    - name: Run Trivy vulnerability scanner
+      uses: aquasecurity/trivy-action@master
+      if: "${{ github.event.inputs.tag == '' }}"
```

Comment:
nit - rather than splitting the trivy call, could split the tag assignment. I'm not sure if it'll allow two steps with the same ID, but writing to github env instead of github output could accomplish it if not.

Reply:
updated.